File: sample-infrastructure-services-domain/npm/src/commands/remove-something.ts
import { inject, injectable } from "inversify";
import { ISampleUnitOfWork, ISampleUnitOfWorkFactory, TYPES as Domain } from "sample-domain";
import { IRemoveSomethingHandler, RemoveSomethingCommand } from "sample-services";
import { DomainHandlerBase } from "../domain-handler";

@injectable()
export class DomainRemoveSomethingHandler extends DomainHandlerBase<RemoveSomethingCommand, number> implements IRemoveSomethingHandler {

    readonly commandType = RemoveSomethingCommand.TYPE;

    constructor(@inject(Domain.Aggregates.ISampleUnitOfWorkFactory) uowFactory: ISampleUnitOfWorkFactory) {
        super(uowFactory);
    }

    async onHandleAsync(uow: ISampleUnitOfWork, command: RemoveSomethingCommand): Promise<number> {
        var parent = await uow.parents.getAsync(command.someId);
        if (!parent) {
            throw new Error(`Could not locate record for id (${command.someId})`);
        }
        await uow.parents.deleteAsync(command.someId);
        return 0;
    }

}

File: sample-infrastructure-services-domain/dotnet/src/commands/remove-something.cs
using System.Collections.Generic;
using System.Threading.Tasks;
using Sample.Domain.Aggregates;
using Sample.Services.Commands;

namespace Sample.Infrastructure.Services.Domain.Commands
{
    public class DomainRemoveSomethingHandler : DomainHandlerBase<RemoveSomethingCommand, int>, IRemoveSomethingHandler
    {
        public DomainRemoveSomethingHandler(ISampleUnitOfWorkFactory uowFactory) : base(uowFactory) { }
        
        public override string CommandType => RemoveSomethingCommand.TYPE;

        protected override async Task<int> OnHandleAsync(ISampleUnitOfWork uow, RemoveSomethingCommand command) 
        {
            var parent = await uow.Parents.GetAsync(command.SomeId);
            if (parent == null)
                throw new KeyNotFoundException($"Could not locate record for id ({command.SomeId})");
            await uow.Parents.DeleteAsync(command.SomeId);
            return 0;
        }
    }
}
